community/graph-algo/src/main/java/org/neo4j/graphalgo/impl/path/ShortestPath.java

```diff
@@ -335,34 +335,44 @@ private void prepareNextLevel()
         @Override
         protected Node fetchNextOrNull()
         {
-            while ( true )
-            {
+            return foundANode();
```

Comment:
I think this whole iterator change may be unnecessary. It should be enough that the filter is called in prepareNextLevel().